base_local_planner/src/obstacle_cost_function.cpp

```diff
@@ -148,13 +140,21 @@ double ObstacleCostFunction::footprintCost (
     const double& y,
     const double& th,
     const std::vector<geometry_msgs::Point>& scaled_footprint,
-    costmap_2d::Costmap2D* costmap,
-    base_local_planner::WorldModel* world_model) {
+    costmap_2d::LayeredCostmap* layered_costmap,
+    double t) {
 
   //check if the footprint is legal
   // TODO: Cache inscribed radius
-  double footprint_cost = world_model->footprintCost(x, y, th, scaled_footprint);
+  costmap_2d::Costmap2D* costmap = layered_costmap->getCostmap(t);
+
+  if (costmap == NULL) {
+    return -1.0; // What to return ???????????
+  }
 
+  std::unique_ptr<base_local_planner::WorldModel> world_model(new base_local_planner::CostmapModel(*costmap));
```

Comment:
```suggestion
  auto world_model = std::make_unique<base_local_planner::WorldModel>(*costmap);
```
F: use `make_unique` in more modern cpp